compiler/modules/bitcell_array.py

```diff
@@ -30,9 +30,18 @@ def __init__(self, cols, rows, name="bitcell_array"):
         self.height = self.row_size*self.cell.height + drc["well_enclosure_active"]
         self.width = self.column_size*self.cell.width 
         
-        self.add_pins()
+        if(OPTS.bitcell == "pbitcell"):
+            self.add_multiport_pins()
+        else:
+            self.add_pins()
+        
         self.create_layout()
-        self.add_layout_pins()
+        
+        if(OPTS.bitcell == "pbitcell"):
```

Comment:
Maybe make this a member function of the bitcell and pass this instance's self reference to add the pins. Then each type of bitcell can add whatever pins it needs.